### PR TITLE
Resizable columns - Insights table

### PR DESCRIPTION
**Pull request by paolodamico**
## Changes

This PR converts the insights table into a table with resizable columns so the user can adjust the width at will for large series. Feedback from users.

image

## How did you test this code?

See recording above. Manually tested by creating a trends insight with "Current URL" as a breakdown.

### CODE REVIEW

File: frontend/src/lib/components/ResizableTable/ResizableTable.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ import './index.scss'
 
 export const TableConfig = _TableConfig
 
-export interface ResizableColumnType<RecordType> extends ColumnType<RecordType> {
+export interface ResizableColumnType<RecordType> extends Omit<ColumnType<RecordType>, 'width'> {
```
Comment (paolodamico) [inline frontend/src/lib/components/ResizableTable/ResizableTable.tsx:16]: `width` property is set dynamically internally, therefore it shouldn't be passed.

File: frontend/src/scenes/insights/InsightsTable/InsightsTable.tsx [hunk 2]
```diff
@@ -84,7 +84,9 @@ export function InsightsTable({ isLegend = true, showTotalCount = false }: Insig
                 )
             },
             fixed: 'left',
-            width: 30,
+            span: 1,
+            defaultWidth: 50,
+            widthConstraints: [50, 50],
```
Comment (paolodamico) [inline frontend/src/scenes/insights/InsightsTable/InsightsTable.tsx:89]: No better way to restrict sizing

File: frontend/src/scenes/insights/InsightsTable/InsightsTable.tsx [hunk 3]
```diff
@@ -139,20 +141,25 @@ export function InsightsTable({ isLegend = true, showTotalCount = false }: Insig
...
+                    />
+                ),
+                render: function RenderPeriod({}, item: IndexedTrendResult) {
+                    return maybeAddCommasToInteger(item.data[index])
+                },
+                sorter: (a, b) => a.data[index] - b.data[index],
+                align: 'center',
+                span: 1,
+                widthConstraints: [96, Infinity],
+                defaultWidth: 150,
```
Comment (paolodamico) [inline frontend/src/scenes/insights/InsightsTable/InsightsTable.tsx:160]: `150px` for some reason translates into `96px` on initial rendering. Any other value causes the columns to jump.

**Comment by paolodamico** [comment]
This implementation is still buggy because the underlying `ResizableTable` has a few issues. Opted to leave this for now as we have plans to improve tables anyways. One big issue is with the last column, and having a right-fixed column. There are in addition some other issues noted within the code. @mariusandra not sure if we plan to pick up tables soon (in which case this would be pretty useful) and if not we can close this.

**Issue with last column:**
image

**Review by mariusandra — CHANGES_REQUESTED** [review]
Since the resizable tables are inherently broken, this PR now breaks the insight tables:



(always, always enables fixed scrollbars on macos if you're a frontend developer)

The screencast is after a full reload of the page. What was previously a table, is now a table with two scrollbars.

Personally I'd remove all resizable tables on all pages, since they:
1. Make the entire page really slow (check how long the "feature flags" page renders if switching between sidebar items... and the flags have already been loaded)
2. Have all the unresolved UX bugs.

There's a task - https://github.com/PostHog/posthog/issues/5361 - which is in scope for @Twixes for this sprint, however with a focus on design first, and all the complicated resizing behaviour only if there's time.

**Comment by paolodamico** [comment]
Will close this for now, if we do add resizing column support and this is helpful, we can reopen.

**Pull request closed without merging**